lib/model/Settings.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:openfoodfacts/utils/QueryType.dart';
+
+class Settings {
```

Comment:
This change makes sense, thanks for adding. Question: Can we later add other settings like language etc to this file ? Currently it's passed into the getUri method. What's the plan for expanding the this file to include more settings, what else will be here etc (just curious).

Reply:
Thanks for your review Jasmeet, exactly I am going to expand that pr later on, but since it's going to bring a lot refactoring code changes I decided to split the PR so that's it stays reviewabls.

The next thing will be a `globalQueryType` and the depreciation of passing them to every method manually and also creating a global user and moving the user agent into the settings.

I am also planning to add some way to choose the queried language from inside the settings but for that I have to find a reliable way which still customizable enough for use in translation apps.

Documentation will be added down the line.

Reply:
monsieurtanuki@ I think the idea is to turn this file into a generic settings/config class, changing these parameters can change the app's behavior and the idea is that a dev can update these parameters to test different behaviors locally. 

Currently it does look like UriHelper but in the future it would turn into AppSettings or AppConfig, we can call it one of those? Any other ideas ?

Reply:
@jasmeet0817 Fair enough but for the moment I cannot visualize other usages of this class: can you please name one - in the next future?

Reply:
@monsieurtanuki 

> 
> The next thing will be a `globalQueryType` and the depreciation of passing them to every method manually and also creating a global user and moving the user agent into the settings.
> 
> I am also planning to add some way to choose the queried language from inside the settings but for that I have to find a reliable way which still customizable enough for use in translation apps.

---------

I myself am not satasfied with the current name, what about something like `OpenFoodAPIClientSettings` or in short `OpenFoodSettings`

Reply:
@M123-dev We're still only talking about the URI, am I right?
Anyway, what about `OpenFoodAPISettings`?
Or, we could even move the code of `Settings` into `OpenFoodAPIClient`: there's no other place where this code could be reused. Not convinced it's a good idea, though, and that would probably thwart my idea of `getInstance()` vs. `static`.

Reply:
So the general idea here (correct me if I'm wrong) is to come up with a set of settings that can change the behavior of the app. For example if you wanted to test the eco score in Switzerland, there is no way of doing it right now except changing the getProduct method and passing the 'cc' query parameter set as 'DE' or "FR' etc, but in the future we can force set the setting for "cc" to allow eco scores to be shown in a local version of the app. This then becomes the poor man's version of flags for the repo.

The reason it makes sense to extract this into a class of it's own rather than inside an existing one like `OpenFoodAPIClient` is because it should be easy for new developers to come and see where all such settings live, so they can easily override something they need to.

I would actually advice not having any methods in this class (although upto you guys), but just a bunch of overridable settings. And then you could have two classes:

1. `OpenFoodAPISettings` with all the settings.
2. (optional - if needed) UriHelper with the helper methods, this can use the variables from `OpenFoodAPISettings` as needed.

What do you guys think?

Reply:
@monsieurtanuki I actually have thought about this when starting, this would also allow us to make getUri private but would again add more code to the already 600 line long OpenFoodAPIClient

Reply:
@jasmeet0817 thats indeed a good idea in terms of overview and abstraction

Reply:
If you ask me this is what I would do. Looks much cleaner to me, but we've already spent a lot of time on the subject, so feel free to ignore this suggestion:
```dart
// default value: prod
OpenFoodAPISettings.instance = OpenFoodAPISettings.fromQueryType(QueryType.PROD);

class OpenFoodAPISettings {
  static OpenFoodAPISettings instance;
  
  OpenFoodAPISettings({
      this.scheme,
      this.host,
      this.robotoffHost,
    });

  OpenFoodAPISettings.fromQueryType(final QueryType queryType)
    : this(
        scheme: _URI_SCHEME_HTTPS,
        host: queryType == QueryType.PROD ? _URI_HOST_PROD : _URI_HOST_TEST,
        robotoffHost: queryType == QueryType.PROD ? _URI_ROBOTOFF_HOST_PROD : _URI_ROBOTOFF_HOST_TEST,
      );

  final String scheme;
  final String host;
  final String robotoffHost;

  static String const _URI_SCHEME_HTTPS = 'https';
  static String const _URI_HOST_PROD = 'world.openfoodfacts.org';
  static String const _URI_HOST_TEST = 'world.openfoodfacts.net';

  static String const _URI_ROBOTOFF_HOST_PROD = 'robotoff.openfoodfacts.org';
  static String const _URI_ROBOTOFF_HOST_TEST = 'robotoff.openfoodfacts.net';

  Uri _getUri({
    String path,
    Map<String, dynamic>? queryParameters,
    bool isRobotoff,
  }) =>
    Uri(
      scheme: scheme,
      host: isRobotoff ? robotoffHost : host,
      path: path,
      queryParameters: queryParameters,
    );
  }

  Uri getUri({
    String path,
    Map<String, dynamic>? queryParameters,
  }) => _getUri(path: path, queryParameters: queryParameters, isRobotoff: false);

  Uri getRobotoffUri({
    String path,
    Map<String, dynamic>? queryParameters,
    }) => _getUri(path: path, queryParameters: queryParameters, isRobotoff: true);
}
```

Reply:
monsieurtanuki@ I agree with most of your code but I still think it would be beneficial (more modular and dev friendly) to abstract out a settings-only class (`OpenFoodAPISettings`) that only contains a bunch of variables that can be changed to modify the app behavior (example: `language`, `country`, `queryType`, `useSecureScheme` etc). And then we can have a `UriHelper` class which would mostly have the code you mention in your comment. It would use `apiSettings.useSecureScheme` and `apiSettings.queryType` inside where `apiSettings = OpenFoodAPISettings.getInstance()`.

For tests you could do `testSettings =OpenFoodAPISettings.getTestInstance()` to return test instance instead.

Example usage: This way when you call a method like `OpenFoodAPIClient.getRobotoffQuestionsForProduct`, you would pass in the `apiSettings` as a parameter, it will then call `uriHelper` with the `apiSettings`. For tests you would pass `testSettings` instead.

Also for example, if you had to run a local instance without https, you would just go to settings file and set useSecureScheme = false. You wouldn't have to deal with any other code which is unrelated to settings. 

Just my 2 cents, I still think you guys have been on the project longer so it should be your call though.